src/Infrastructure/Event/Telemetry/TelemetryConsoleEventListener.php

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Appplication\Service\Telemetry\TelemetryEventsSynchronizerInterface $telemetryEventsSynchronizer
+     * @param \SprykerSdk\Sdk\Core\Appplication\Service\Telemetry\TelemetryEventMetadataFactoryInterface $telemetryEventMetadataFactory
+     * @param \SprykerSdk\Sdk\Infrastructure\Event\Telemetry\TelemetryConsoleEventValidatorInterface $telemetryConsoleEventValidator
+     * @param bool $isTelemetryEnabled
+     */
+    public function __construct(
+        TelemetryEventsSynchronizerInterface $telemetryEventsSynchronizer,
+        TelemetryEventMetadataFactoryInterface $telemetryEventMetadataFactory,
+        TelemetryConsoleEventValidatorInterface $telemetryConsoleEventValidator,
+        bool $isTelemetryEnabled
+    ) {
+        $this->telemetryEventsSynchronizer = $telemetryEventsSynchronizer;
+        $this->telemetryEventMetadataFactory = $telemetryEventMetadataFactory;
+        $this->telemetryConsoleEventValidator = $telemetryConsoleEventValidator;
+        $this->isTelemetryEnabled = $isTelemetryEnabled;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Event\ConsoleTerminateEvent $event
+     *
+     * @return void
+     */
+    public function onConsoleTerminate(ConsoleTerminateEvent $event): void
+    {
+        if (!$this->isTelemetryEnabled || !$this->telemetryConsoleEventValidator->isValid($event)) {
+            return;
+        }
+
+        $this->addSuccessfulCommandEvent($event);
+
+        $this->synchronizeEvents($event);
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Event\ConsoleErrorEvent $event
+     *
+     * @return void
+     */
+    public function onConsoleError(ConsoleErrorEvent $event): void
+    {
+        if (!$this->isTelemetryEnabled || !$this->telemetryConsoleEventValidator->isValid($event)) {
+            return;
+        }
+
+        $this->addFailedCommandEvent($event);
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Event\ConsoleTerminateEvent $event
+     *
+     * @return void
+     */
+    protected function addSuccessfulCommandEvent(ConsoleTerminateEvent $event): void
+    {
+        $telemetryEvent = new TelemetryEvent(new CommandExecutionPayload(
+            (string)($event->getCommand() !== null ? $event->getCommand()->getName() : ''),
+            $event->getInput()->getArguments(),
+            $event->getInput()->getOptions(),
+            '',
+            $event->getExitCode(),
+        ), $this->telemetryEventMetadataFactory->createTelemetryEventMetadata());
+
+        $this->telemetryEventsSynchronizer->persist($telemetryEvent);
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Event\ConsoleErrorEvent $event
+     *
+     * @return void
+     */
+    protected function addFailedCommandEvent(ConsoleErrorEvent $event): void
+    {
+        $telemetryEvent = new TelemetryEvent(new CommandExecutionPayload(
+            (string)($event->getCommand() !== null ? $event->getCommand()->getName() : ''),
```

Comment:
we don't need to trigger event if the command name is empty string. it's unuseful information